openpgp/packet/notation.go

```diff
@@ -0,0 +1,29 @@
+package packet
+
+// Notation type represents a Notation Data subpacket
+// see https://tools.ietf.org/html/rfc4880#section-5.2.3.16
+type Notation struct {
+	Name string
+	Value []byte
+	IsCritical bool
+	IsHumanReadable bool
+}
+
+func (not *Notation) getData() []byte {
+	nameData := []byte(not.Name)
+	nameLen := len(nameData)
+	valueLen := len(not.Value)
+
+	data := make([]byte, 8 + nameLen + valueLen)
+	if not.IsHumanReadable {
+		data[0] = 0x80
+	}
+
+	data[4] = byte(nameLen >> 8)
+	data[5] = byte(nameLen)
+	data[6] = byte(valueLen >> 8)
+	data[7] = byte(valueLen)
+
+	data = append(data, nameData...)
```

Comment:
I think append would not work as intended here.
Data is already set to have length = 8 + nameLen + valueLen, and append will write after those bytes, not overwrite them.
I think what we want is to use `copy` here.
See https://github.com/ProtonMail/go-crypto/pull/144